### ISSUE TITLE
Broken example under 'Changing Containers' in documentation

### ISSUE THREAD

**Issue by cheryllium** [react-bootstrap/react-bootstrap#6009]
## Describe the bug

In the embedded code example for Changing Containers in the documentation, clicking the 'Holy Guacamole' button causes following error: 

```
InternalError: too much recursion
```

## To Reproduce

1. Under Components, click Overlays
2. Click or scroll down to "Changing Containers" section
3. In the example, click button 'Holy Guacamole'

This is reproducible on the following URLs: 
- https://react-bootstrap.github.io/components/overlays/#overlays-container
- https://react-bootstrap.netlify.app/components/overlays/#overlays-container

## Expected behavior

Display example popover on button click.

## Screenshots

Screenshot: 

![image](https://user-images.githubusercontent.com/5027241/131692336-b060ad71-19ff-4b68-b9c2-938990a963ad.png)

## Environment (please complete the following information)

Reproduced on: 
- NixOS + Firefox 83.0 64-bit
- Windows 10 + Firefox 91.0.2 64-bit

**Comment by kyletsang** [comment]
Took a brief look at this and looks like it's an issue with running `dequal` on popper modifiers with html elements.  For example the arrow modifier is one of these

https://popper.js.org/docs/v2/modifiers/arrow/

cc @jquense

**Comment by kyletsang** [comment]
@cheryllium to get this to work, replace `container={ref.current}` with `container={ref}` in the above example in docs.

**Comment by jquense** [comment]
interesting! so the issue is that it's comparing elements which fails?

**Comment by kyletsang** [comment]
> 
> 
> interesting! so the issue is that it's comparing elements which fails?

Yeah, I guess there's a circular reference somewhere within the element?   Can we extract the modifiers with elements and use `isEqualNode` on them?

**Comment by cheryllium** [comment]
Thank you very much! Cheers!
